docs/source/development/documentation/documentation.md

```diff
@@ -0,0 +1,153 @@
+automatically runs any code in MyST markdown documents and renders the resulting code
+and outputs into HTML. The documentation can therefore contain dynamically generated
+examples of running code and output plots to demonstrate how to use the
+package. This build process also ensures that code in documentation is kept up to date
+with the package. The syntax for including code blocks in documents is shown below -
+this example includes a tag that causes the code to be concealed in the built
+documentation, but with a toggle button to allow it to be expanded.
+
+````{code-block}
+```{code-cell} python
+:tags: [hide-input]
+# This is just an example code cell to demonstrate how code is included in
+# the pyrealm documentation.
+```
+````
+
+### Table of contents
+
+We use the `sphinx_external_toc` package to maintain a table of contents for the
+package. The file `docs/source/_toc.yml` contains the structure of the table and you
+will need to add new documentation files to this file for them to appear in the table.
+The documentation build process will fail if it finds files in `docs/source` that are
+not included in the table of contents!
+
+### Docstrings
+
+The `pyrealm` package uses docstrings written in the [Google
+style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+This allows the function documentation to be stored alongside the code and it is included
+in the documentation using the `sphinx` `autodoc` extension. See the code itself for
+examples of the documentation formatting and typical content.
+
+At the moment, we use the `autodoc` plugins for `sphinx` to convert docstrings to HTML
+and build the online API documentation. Unfortunately, the `autodoc` package is
+hard-coded to expect docstrings to use reStructuredText, which means that at the moment
+**all docstrings have to be written in `rst` format**. At some point, we'd like to
+switch away to using Markdown throughout, but for the moment look at the existing
+docstrings to get examples of how the formatting differs.
+
+### Referencing
+
+Both the `docs/source` and docstrings uses the `sphinxcontrib-bibtex` package to support
+citations. This uses Latex like citation keys in the documentation to insert references
+and build a bibliography. The `sphinx` configuration in `docs/source/conf.py` provides a
+custom Author/Year citation style. The reference library in `source/refs.bib` needs to
+be kept up to date with the literature for the project.
+
+The three common use cases are shown below using a couple of reference tags
+(`campbell_introduction_2012` and `porporato_hydrologic_2003`) that are included
+in the current [reference library](../../bibliography.md).
+
+* Cite with date in parentheses (``{cite:t}`campbell_introduction_2012` ``): the model
+  implemented in {cite:p}`campbell_introduction_2012`.
+* Cite with reference(s) in parentheses
+  (``{cite:p}`campbell_introduction_2012,porporato_hydrologic_2003` ``): using the P
```

Comment:
I assume this is also a relevant reference for the VE

Reply:
I've swapped the original refs for something alraedy in our docs - more or less at random, so these could be better!